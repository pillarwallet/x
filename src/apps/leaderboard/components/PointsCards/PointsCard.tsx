// utils
import { formatAmountDisplay } from '../../../../utils/number';

// components
import BodySmall from '../Typography/BodySmall';

type PointsCardProps = {
  title: string;
  icon: string;
  background: string;
  points: number;
  rank?: number;
  textTooltip?: string;
};

const PointsCard = ({
  title,
  icon,
  background,
  points,
  rank,
  textTooltip,
}: PointsCardProps) => {
  return (
    <div
      className={`group flex flex-col w-full gap-3 rounded-[10px] p-2 ${background} relative`}
    >
      {textTooltip && (
        <div className="absolute bottom-full mb-2 left-0 rounded-lg border border-white/[.05] bg-[#25232D] px-2.5 py-2 text-[10px] text-white italic font-normal opacity-0 group-hover:opacity-100 transition-all duration-200 z-10 max-w-max w-fit origin-bottom scale-y-0 group-hover:scale-y-100 transform">
          {textTooltip}
        </div>
      )}
      <div className="flex gap-1.5 items-center">
        <div className="flex h-6 w-6 items-center justify-center rounded-[5px]">
          <img src={icon} alt={`${title}-icon`} />
        </div>
        <BodySmall className="font-normal text-white">{title}</BodySmall>
      </div>
      <div className="flex w-full items-center justify-between px-1">
        <BodySmall className="font-normal text-white">Points</BodySmall>
        <BodySmall className="font-semibold text-white">
          {formatAmountDisplay(Math.floor(points || 0))}{' '}
          <span className="text-white/[.5]">PX</span>
        </BodySmall>
      </div>
      <div className="flex w-full items-center justify-between px-1 mb-1">
        <BodySmall className="font-normal text-white">Rank</BodySmall>
        <BodySmall className="font-semibold text-white">
          <span className="text-white/[.5]">#</span>
          {rank !== undefined && rank > 0 ? rank : 'â€“'}
        </BodySmall>
      </div>
    </div>
  );
};

export default PointsCard;
