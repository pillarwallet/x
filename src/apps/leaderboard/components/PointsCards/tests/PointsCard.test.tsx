import { fireEvent, render, screen } from '@testing-library/react';

// components
import PointsCard from '../PointsCard';

describe('<PointsCard />', () => {
  const mockPointsCardProps = {
    title: 'Trading',
    icon: 'trading-icon.png',
    background: 'bg-blue-500',
    points: 1234,
    rank: 1,
    textTooltip: 'This is a tooltip',
  };

  it('renders correctly and matches snapshot', () => {
    const { container } = render(
      <PointsCard
        title={mockPointsCardProps.title}
        icon={mockPointsCardProps.icon}
        background={mockPointsCardProps.background}
        points={mockPointsCardProps.points}
        rank={mockPointsCardProps.rank}
        textTooltip={mockPointsCardProps.textTooltip}
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders title, icon, points and rank correctly', () => {
    render(
      <PointsCard
        title={mockPointsCardProps.title}
        icon={mockPointsCardProps.icon}
        background={mockPointsCardProps.background}
        points={mockPointsCardProps.points}
        rank={mockPointsCardProps.rank}
        textTooltip={mockPointsCardProps.textTooltip}
      />
    );

    expect(screen.getByText('Trading')).toBeInTheDocument();

    const icon = screen.getByAltText('Trading-icon');
    expect(icon).toBeInTheDocument();
    expect(icon).toHaveAttribute('src', 'trading-icon.png');

    expect(screen.getByText(/1234/i)).toBeInTheDocument();
    expect(screen.getByText('PX')).toBeInTheDocument();

    expect(screen.getByText('1')).toBeInTheDocument();
  });

  it('shows "–" for rank if rank is undefined', () => {
    render(
      <PointsCard
        title={mockPointsCardProps.title}
        icon={mockPointsCardProps.icon}
        background={mockPointsCardProps.background}
        points={mockPointsCardProps.points}
        rank={undefined}
        textTooltip={mockPointsCardProps.textTooltip}
      />
    );
    expect(screen.getByText('–')).toBeInTheDocument();
  });

  it('renders tooltip only when textTooltip prop is provided', () => {
    render(
      <PointsCard
        title={mockPointsCardProps.title}
        icon={mockPointsCardProps.icon}
        background={mockPointsCardProps.background}
        points={mockPointsCardProps.points}
        rank={mockPointsCardProps.rank}
        textTooltip={mockPointsCardProps.textTooltip}
      />
    );

    const tooltip = screen.getByText('This is a tooltip');
    expect(tooltip).toBeInTheDocument();
    expect(tooltip).toHaveClass('opacity-0');

    fireEvent.mouseOver(screen.getByText('Trading'));
  });

  it('displays zero points correctly', () => {
    render(
      <PointsCard
        title={mockPointsCardProps.title}
        icon={mockPointsCardProps.icon}
        background={mockPointsCardProps.background}
        rank={mockPointsCardProps.rank}
        textTooltip={mockPointsCardProps.textTooltip}
        points={0}
      />
    );
    expect(screen.getByText('0')).toBeInTheDocument();
    expect(screen.getByText('PX')).toBeInTheDocument();
  });
});
