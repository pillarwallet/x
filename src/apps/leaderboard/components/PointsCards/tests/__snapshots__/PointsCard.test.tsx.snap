// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PointsCard /> renders correctly and matches snapshot 1`] = `
<div>
  <div
    class="flex flex-col w-full gap-3 rounded-[10px] p-2 bg-blue-500"
  >
    <div
      class="relative w-full group flex gap-1.5 items-center"
    >
      <div
        class="flex h-6 w-6 items-center justify-center rounded-[5px]"
      >
        <img
          alt="Trading-icon"
          src="trading-icon.png"
        />
      </div>
      <div
        class="absolute bottom-full mb-2 left-0 rounded-lg border border-white/[.05] bg-[#25232D] px-2.5 py-2 text-[10px] text-white italic font-normal opacity-0 group-hover:opacity-100 transition-all duration-200 z-10 w-full origin-bottom scale-y-0 group-hover:scale-y-100 transform"
      >
        This is a tooltip
      </div>
      <p
        class="text-sm font-medium font-normal text-white"
      >
        Trading
      </p>
    </div>
    <div
      class="flex w-full items-center justify-between px-1"
    >
      <p
        class="text-sm font-medium font-normal text-white"
      >
        Points
      </p>
      <p
        class="text-sm font-medium font-semibold text-white"
      >
        1234
         
        <span
          class="text-white/[.5]"
        >
          PX
        </span>
      </p>
    </div>
    <div
      class="flex w-full items-center justify-between px-1 mb-1"
    >
      <p
        class="text-sm font-medium font-normal text-white"
      >
        Rank
      </p>
      <p
        class="text-sm font-medium font-semibold text-white"
      >
        <span
          class="text-white/[.5]"
        >
          #
        </span>
        1
      </p>
    </div>
  </div>
</div>
`;
